docs/current/sdk/go/guides/core-concepts/710884-private-repositories.md

```diff
@@ -0,0 +1,67 @@
+---
+slug: /sdk/go/710884/private-repositories
+displayed_sidebar: 'current'
+---
+
+# How do I login to clone Git repositories ?
+
+Dagger forwards the `SSH authentication agent` as a secure way to authenticate against Git repositories.
+
+On your computer, please run `ssh-add` to add your ssh key to the authentication agent. This is the only requirement to clone private repositories.
+
+## How to reproduce ?
+
```

Comment:
To see how this works, assume that you have a Dagger CI tool containing the following code, which references a private repository:

INSERT CODE HERE

Now, first remove all the SSH keys from the authentication agent:

```shell
ssh-add -D
All identities removed.
```

Attempt to run the Go CI tool:

```shell
go run .
panic: input:1: git.branch.tree.file.contents failed to load cache key: failed to fetch remote 
exit status 128
```

The tool fails, as it is unable to find the necessary authentication credentials in the SSH authentication agent.

Now, add the SSH key to the authentication agent and try again:

INSERT CODE HERE

```shell
ssh-add
Identity added: xxxxx
go run .  
readme #
```

In this case, the tool succeeds in accessing the private Git repository.